File: event-archive/src/main/java/com/omnibus/archive/model/RuleUpdateByDay.java
package com.omnibus.archive.model;

import java.time.Instant;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.Table;

@Table("rule_updates_by_day")
public class RuleUpdateByDay {

    @PrimaryKey
    private RuleUpdateByDayKey key;

    private Instant occurredAt;
    private String ruleName;
    private String sourcePath;
    private Boolean managed;
    private Integer contentLength;

    public RuleUpdateByDay() {
    }

    public RuleUpdateByDay(
        RuleUpdateByDayKey key,
        Instant occurredAt,
        String ruleName,
        String sourcePath,
        Boolean managed,
        Integer contentLength
    ) {
        this.key = key;
        this.occurredAt = occurredAt;
        this.ruleName = ruleName;
        this.sourcePath = sourcePath;
        this.managed = managed;
        this.contentLength = contentLength;
    }

    public RuleUpdateByDayKey getKey() {
        return key;
    }
}

File: services/legacy/event-archive/src/main/kotlin/pl/writeonly/omnibus/archive/model/RuleUpdateByDay.kt
package pl.writeonly.omnibus.archive.model

import org.springframework.data.cassandra.core.mapping.PrimaryKey
import org.springframework.data.cassandra.core.mapping.Table
import java.time.Instant

@Table("rule_updates_by_day")
data class RuleUpdateByDay(
    @PrimaryKey
    val key: RuleUpdateByDayKey,
    val occurredAt: Instant,
    val ruleName: String,
    val sourcePath: String,
    val managed: Boolean,
    val contentLength: Int,
)

